precompiles/wasmd/wasmd.go

```diff
@@ -373,11 +374,15 @@ func (p Precompile) executeBatch(ctx sdk.Context, method *abi.Method, caller com
 			return
 		}
 
+		ctx = ctx.WithEvmEventManager(sdk.NewEVMEventManager())
 		res, err := p.wasmdKeeper.Execute(ctx, contractAddr, senderAddr, msg, coins)
 		if err != nil {
 			rerr = err
 			return
 		}
+		for _, e := range ctx.EVMEventManager().Events() {
```

Comment:
at some point I'd like to refactor this stuff into some kind of 'post handler' and 'pre handler' that is shared across all the scenarios so we can avoid the duplication.